packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/utils/normalizeRelations.js

```diff
@@ -26,12 +26,21 @@ export const normalizeRelations = (
   relations,
   { modifiedData = {}, shouldAddLink = false, mainFieldName, targetModel }
 ) => {
+  // To display oldest to newest relations we need to reverse each elements in array of results
+  // and reverse each arrays itself
+  const existingRelationsReversed = [...(relations?.data?.pages ?? [])]
+    .map((relation) => ({
+      ...relation,
+      results: relation.results.slice().reverse(),
+    }))
+    .reverse();
+
   return {
     ...relations,
     data: {
       pages:
         [
-          ...(relations?.data?.pages ?? []),
+          ...(targetModel ? existingRelationsReversed : relations?.data?.pages ?? []),
```

Comment:
What's `targetModel`?

Reply:
it's the entity/entry model, it exists only if we are in a relation (to add a link to this entry) not in a search action
if we go for this logic I think we could pass more specific information to know if we are in relations or search, but I'm still unsure about it overall

Reply:
I replaced it with `isSearch` for now because I'm not comfortable with `targetModel` - I'm still wondering if there would be a way to apply those transformations to relations but not on search without making the function dependent on the context of usage 🤷🏻‍♀️

Reply:
The only way of the top of my head would be to make a make a separate function that handles the reversing and then pipe those functions together when you need them (a bit of functional programming). However, @gu-stav may be able to advise better than I!

Reply:
Question here! Why do you work with an array of arrays and not flatten it directly?

I see in RElationInputDataManager.js [L82](https://github.com/strapi/strapi/blob/4e917068b1cd8a5f8d0ecf429c844169ce40dbd6/packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/RelationInputDataManager.js#L82), you flat the data, why not reverse it there?

Reply:
Also, why not making another function to normalize relations for the search?

Why having a function with multiple responsibilities here?

Reply:
> I see in RElationInputDataManager.js [L82](https://github.com/strapi/strapi/blob/4e917068b1cd8a5f8d0ecf429c844169ce40dbd6/packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/RelationInputDataManager.js#L82), you flat the data, why not reverse it there?

in L82 we already have normalized the relations, meaning:
- saved and not saved relations are merged
- they are filtered (depending on relations being in disconnect array or not)

so L82 I get something like:
```
[
    {
      results: [
        { id: 1, label: 'newest saved' },
        { id: 2, label: 'second saved' },
      ],
    },
    {
      results: [
        { id: 3, label: 'third saved' },
        { id: 4, label: 'oldest saved' },
      ],
    },
    {
      results: [
        { label: 'oldest not saved' },
        { label: 'newest not saved' },
      ],
    },
]
```

The first 2 arrays (coming from db) need to be reversed inside, and both of them need to be reversed outside (hard to explain 😅)
The last one shouldn't be reversed at all (not the items inside, and not the array itself)

does that make sense?

> Why having a function with multiple responsibilities here?

good question, on one side it is Gustav's work and I didn't want to break everything while he couldn't review it
at first I think there wasn't that much complexity forcing to duplicate some part of the logic, but with the reverse logic to apply maybe we should 🤷🏻‍♀️

Reply:
> The last one shouldn't be reversed at all (not the items inside, and not the array itself)
💡 Got it! Thank you

>  but with the reverse logic to apply maybe we should
agree

Reply:
I agree with what has been said: I think this is enough justification for creating a new method for search results (`normalizeRelations()` and `normalizeSearchResults()`). Both of them can share how a single relation is normalized, but since they do different things in terms of the order, I'd say it makes sense to split.

I also think we should move the reverse order within a page into the `useRelation` hook and transform the data using `select`: https://tanstack.com/query/v4/docs/guides/infinite-queries#what-if-i-want-to-show-the-pages-in-reversed-order

Wdyt?

Reply:
sounds great @gu-stav, thank you for the doc, on it!

Reply:
done [here](https://github.com/strapi/strapi/pull/14527/commits/359b4e756eca71472b313f66101ea25b1d2f7e8f)
I added test about reverse in `normalizeRelations` and I'll add quick tests on `normalizeSearchResults`
I'm stuck with `useRelations` hooks tests for a big part of the day - I commented the tests breaking for now